### PR TITLE
feat: add article for JavaScript String.valueOf()

### PR DESCRIPTION
**Pull request by HasanAbdullah31**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md [hunk 1]
```diff
@@ -3,13 +3,18 @@ title: String.prototype.valueOf
...
 ## String.prototype.valueOf
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/javascript/standard-objects/string/string-prototype-valueof/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+The `valueOf()` method returns the primitive string value of the given String object.
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+**Example**
+```js
+var x = new String("hi");
+console.log(x);        // String {"hi"}
```
Comment (thecodingaviator) [inline guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md]:
```suggestion
console.log(x); // String {"hi"}
```

File: guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md [hunk 2]
```diff
@@ -3,13 +3,18 @@ title: String.prototype.valueOf
...
+console.log(typeof x); // object
+console.log(x.valueOf());        // hi
```
Comment (thecodingaviator) [inline guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md]:
```suggestion
console.log(x.valueOf()); // hi
```

File: guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md [hunk 3]
```diff
@@ -3,13 +3,18 @@ title: String.prototype.valueOf
...
+console.log(typeof x.valueOf()); // string
+```
 
-<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
+*Note*: for String objects, `valueOf()` and `toString()` return the same thing.
```
Comment (thecodingaviator) [inline guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md]:
```suggestion
*Note*: For String objects, `valueOf()` and `toString()` return the same thing.
```

File: guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md [hunk 4]
```diff
@@ -3,13 +3,18 @@ title: String.prototype.valueOf
 ---
 ## String.prototype.valueOf
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/javascript/standard-objects/string/string-prototype-valueof/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+The `valueOf()` method returns the primitive string value of the given String object.
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+**Example**
```
Comment (thecodingaviator) [inline guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md]:
```suggestion
**Usage**
```

File: guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md [hunk 5]
```diff
@@ -3,13 +3,18 @@ title: String.prototype.valueOf
 ---
 ## String.prototype.valueOf
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/javascript/standard-objects/string/string-prototype-valueof/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+The `valueOf()` method returns the primitive string value of the given String object.
```
Comment (thecodingaviator) [inline guide/english/javascript/standard-objects/string/string-prototype-valueof/index.md]:
```suggestion
The `valueOf()` method returns the primitive string value of the given `String` object.
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]

**Comment by HasanAbdullah31** [comment]
@thecodingaviator I added the changes you requested, check the file again.

**Review by thecodingaviator — APPROVED** [review]
LGTM

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**